File: src/DesignPattern/Creational/SimpleFactory/SimpleFactory-go/src/color/color.go
package color

import "fmt"

type Color interface {
	PrintColor()
}

type Red struct{}

func (p *Red) PrintColor() {
	fmt.Println("red")
}


type Black struct{}

func (p *Black) PrintColor() {
	fmt.Println("black")
}

// 设计模式：创建型-简单工厂
// 算法：——
// 复杂度：——
func CreateColor(arg string) Color {
	switch arg {
	case "red":
		return &Red{}
	case "black":
		return &Black{}
	}

	return nil
}

File: src/DesignPattern/Creational/SimpleFactory/SimpleFactory-ts/src/color/Color.ts

interface IColor {
    getColor(): void;
}

class red implements IColor {
    public getColor(): void {
        console.log("red");
    }
}

class black implements IColor {
    public getColor(): void {
        console.log("black");
    }
}

enum ColorTypes {
    red, black,
}

class ColorFactory {

    public static createColor(t: ColorTypes): IColor {
        switch (t) {
            case ColorTypes.red:
                return new red();
            case ColorTypes.black:
                return new black();
        }
        return null;
    }
}

export {
    ColorFactory,
    ColorTypes,
    IColor,
}
